packages/std/src/ibc.rs

```diff
@@ -59,23 +59,52 @@ pub struct IbcEndpoint {
 /// two timeout fields we ensure that at least one timeout is set.
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
-pub enum IbcTimeout {
-    /// Block timestamp (nanoseconds since UNIX epoch) after which the packet times out
-    /// (measured on the remote chain).
-    Timestamp(Timestamp),
-    /// Block after which the packet times out (measured on remote chain).
-    Block(IbcTimeoutBlock),
-    /// Use this to set both timestamp and block timeout. The package then times out once
-    /// the first of both timeouts is hit.
-    Both {
-        timestamp: Timestamp,
-        block: IbcTimeoutBlock,
-    },
+pub struct IbcTimeout {
+    // use private fields to enforce the use of constructors, which ensure that at least one is set
```

Comment:
Ah, this is a clever approach.
Yet unlike Go, these private fields do get JSON serialized

Reply:
Jupp. And this is also how you can hack around the constructors: Deserialize into `IbcTimeout`. But I think this is a fine trade-off.